README.md
## KCFinder
#### A PHP-based AJAX file and media manager

#### Credits

Original project by Pavel Tzonkov http://kcfinder.sunhater.com

#### Description

KCFinder is a web-based file manager, built on a PHP backend and AJAX-powered
UI. It has file upload features, path locking for different file types (when
integrated within various WYSIWYG editors).

#### Requirements

* (server) PHP 5.3+
* (server) (optional) PHP-GD
* (client) any modern web browser
* (client) JavaScript turned on

#### Features

* File system manipulation (move||copy||delete files or directories)
* Thumbnail generation for images (.thumbs directory in the upload path)
* Path locking for file types - specify a path for every filetype you need to
  upload in a separate location (i.e. *PNG*,*JPG*,*GIF* stay in **images** directory,
  *WAV*,*OGG*,*MP3* files in **audio** directory and so on, on a config-based
  behaviour).
* Easy (and documented) integration with a number of WYSIWYG editors (i.e. CKEditor, FCKeditor,
  TinyMCE etc.)
* Ajax engine with JSON responses
* Multiple files upload (does not work in IE!)
* Upload files using HTML5 drag and drop (for Firefox and Chrome only!)
* Download multiple files or a folder as single ZIP file
* Select multiple files with the Ctrl/Command key
* Clipboard for copying, moving and downloading multiple files
* Easy to integrate and configure in web applications
* Option to select and return several files. For custom applications only
* Resize uploaded images. Configurable maximum image resolution
* Configurable thumbnail resolution
* Visual CSS themes
* Multilanguage system
* Slideshow using arrow keys

#### Documentation

For now, the only documentation is available at the official web page,
http://kcfinder.sunhater.com - will be expanded as this fork goes on, and maybe
integrated in the local wiki system.

#### Further development

Right now, the GitHub repository serves as a branch of the original project,
aiming to send patches and features upstream. If, for any reason, they will not
be integrated or ignored, there is a possibility of complete forking and
renaming, as it will become a full-featured fork.

#### Roadmap

The features I aim for right now are (not necessarily in this order):

* Image editing (crop, resize, rotate) from within the KCFinder window
* Code editing plugin with syntax highlight (enabling it to be used as a small
  multipurpose tool)
* Support for mounting (remote) FTP locations

#### Repository and workflow

This repository follows the git flow model (https://github.com/nvie/gitflow) and
uses Vincent Driessen's branching recommendations. Thus, the main stable branch
is named **master**, release is prepared on **release** branches, active
development is taking place on **develop** branch and feature branches are
prefixed with **feature/<name>**.  More details in Vincent's article:
http://nvie.com/git-model



doc/Changelog
VERSION 2.51 - 2010-09-
-------------------------


VERSION 2.5 - 2010-08-23
------------------------
* Drupal module support
* Drag and drop uploading plugin
* Two more language labels
* Localhost cookies bugfix
* Renaming current folder bugfix
* Small bugfixes


VERSION 2.41 - 2010-07-24
-------------------------
* Directory types engine improvement
* New 'denyExtensionRename' config setting added


VERSION 2.4 - 2010-07-20
------------------------
* Online checking if new version is released in About box. To use this
  feature you should to have Curl, HTTP or Socket extension, or
  allow_url_fopen ini setting should be "on"
* New 'denyUpdateCheck' config setting added
* New 'dark' theme added (made by Dark Preacher)
* Additional 'theme' GET parameter to choose a theme from URL
* Thumbnails loading improvement
* Some changes in Oxygen CSS theme
* Replace alert() and confirm() JavaScript functions with good-looking boxes
* Safari 3 right-click fix
* Small bugfixes


VERSION 2.32 - 2010-07-11
-------------------------
* 'filenameChangeChars' and 'dirnameChangeChars' config settings added
* Content-Type header fix for css.php, js_localize.php and
  js/browser/joiner.php
* CKEditorFuncNum with index 0 bugfix
* Session save handler example in core/autoload.php


VERSION 2.31 - 2010-07-01
-------------------------
* Proportional uploaded image resize bugfix
* Slideshow bugfixes
* Other small bugfixes


VERSION 2.3 - 2010-06-25
------------------------
* Replace XML Ajax responses with JSON
* Replace old 'readonly' config option with advanced 'access' option
  PLEASE UPDATE YOUR OLD CONFIG FILE!!!
* Slideshow images in current folder using arrow keys
* Multipe files upload similar to Facebook upload (not works in IE!)
* Option to set protocol, domain and port in 'uploadURL' setting
* Bugfixes


VERSION 2.21 - 2010-11-19
-------------------------
* Bugfixes only


VERSION 2.2 - 2010-07-27
------------------------
* Many bugfixes
* Read-only config option


VERSION 2.1 - 2010-07-04
------------------------
* Endless JavaScript loop on KCFinder disable bugfix
* New config setting whether to generate .htaccess file in upload folder
* Upload to specified folder from CKEditor & FCKeditor direct upload dialog
* Select multiple files bugfixes


VERSION 2.0 - 2010-07-01
------------------------
* Brand new core
* Option to resize files/folders panels with mouse drag
* Select multiple files with Ctrl key
* Return list of files to custom integrating application
* Animated folder tree
* Directory Type specific configuration settings
* Download multiple files or a folder as ZIP file


VERSION 1.7 - 2010-06-17
------------------------
* Maximize toolbar button
* Clipboard for copying and moving multiple files
* Show warning if the browser is not capable to display KCFinder
* Google Chrome Frame support for old versions of Internet Explorer


VERSION 1.6 - 2010-06-02
------------------------
* Support of Windows Apache server
* Support of Fileinfo PHP extension to detect mime types (*mime directory type)
* Option to deny globaly some dangerous extensions like exe, php, pl, cgi etc
* Check for denied file extension on file rename
* Disallow to upload hidden files (with names begins with .)
* Missing last character of filenames without extension bugfix
* Some small bugfixes


VERSION 1.5 - 2010-05-30
------------------------
* Filenames with spaces download bugfix
* FCKEditor direct upload bugfix
* Thumbnail generation bugfixes


VERSION 1.4 - 2010-05-24
------------------------
* Client-side caching bugfix
* Custom integrations - window.KCFinder.callBack()
* Security fixes


VERSION 1.3 - 2010-05-06
------------------------
* Another session bugfix. Now session configuratin works!
* Show filename by default bugfix
* Loading box on top right corner


VERSION 1.2 - 2010-05-03
------------------------
* Thumbnail generation bugfix
* Session bugfix
* other small bugfixes


config.php
<?php

/** This file is part of KCFinder project
  *
  *      @desc Base configuration file
  *   @package KCFinder
  *   @version 2.52-dev
  *    @author Pavel Tzonkov <pavelc@users.sourceforge.net>
  * @copyright 2010, 2011 KCFinder Project
  *   @license http://www.opensource.org/licenses/gpl-2.0.php GPLv2
  *   @license http://www.opensource.org/licenses/lgpl-2.1.php LGPLv2
  *      @link http://kcfinder.sunhater.com
  */

// IMPORTANT!!! Do not remove uncommented settings in this file even if
// you are using session configuration.
// See http://kcfinder.sunhater.com/install for setting descriptions

$_CONFIG = array(

    'disabled' => false,
    'denyZipDownload' => false,
    'denyUpdateCheck' => false,
    'denyExtensionRename' => false,

    'theme' => "oxygen",

    'uploadURL' => "/upload",
    'uploadDir' => "",

    'dirPerms' => 0755,
    'filePerms' => 0644,

    'access' => array(

        'files' => array(
            'upload' => true,
            'delete' => true,
            'copy' => true,
            'move' => true,
            'rename' => true
        ),

        'dirs' => array(
            'create' => true,
            'delete' => true,
            'rename' => true
        )
    ),

    'deniedExts' => "exe com msi bat php phps phtml php3 php4 cgi pl",

    'types' => array(

        // CKEditor & FCKEditor types
        'files'   =>  "",
        'flash'   =>  "swf",
        'images'  =>  "*img",

        // TinyMCE types
        'file'    =>  "",
        'media'   =>  "swf flv avi mpg mpeg qt mov wmv asf rm",
        'image'   =>  "*img",
    ),

    'filenameChangeChars' => array(/*
        ' ' => "_",
        ':' => "."
    */),

    'dirnameChangeChars' => array(/*
        ' ' => "_",
        ':' => "."
    */),

    'mime_magic' => "",

    'maxImageWidth' => 0,
    'maxImageHeight' => 0,

    'thumbWidth' => 100,
    'thumbHeight' => 100,

    'thumbsDir' => ".thumbs",

    'jpegQuality' => 90,

    'cookieDomain' => "",
    'cookiePath' => "",
    'cookiePrefix' => 'KCFINDER_',

    // THE FOLLOWING SETTINGS CANNOT BE OVERRIDED WITH SESSION CONFIGURATION
    '_check4htaccess' => true,
    //'_tinyMCEPath' => "/tiny_mce",

    '_sessionVar' => &$_SESSION['KCFINDER'],
    //'_sessionLifetime' => 30,
    //'_sessionDir' => "/full/directory/path",

    //'_sessionDomain' => ".mysite.com",
    //'_sessionPath' => "/my/path",
);

?>